tests/parser.rs

```diff
@@ -96,12 +99,17 @@ where
     fn poll_read(
         mut self: Pin<&mut Self>,
         cx: &mut task::Context,
-        buf: &mut [u8],
-    ) -> Poll<io::Result<usize>> {
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<io::Result<()>> {
         match self.ops.next() {
             Some(PartialOp::Limited(n)) => {
-                let len = std::cmp::min(n, buf.len());
-                Pin::new(&mut self.inner).poll_read(cx, &mut buf[..len])
+                let len = std::cmp::min(n, buf.remaining());
+                buf.initialize_unfilled();
+                let mut sub_buf = buf.take(len);
+                ready!(Pin::new(&mut self.inner).poll_read(cx, &mut sub_buf))?;
+                let initialized = sub_buf.initialized().len();
+                buf.advance(initialized);
```

Comment:
Nice catch! I think it never shows up in tests because, as there isn't something else reading into the `ReadBuf` before this call it is always empty at this point so the `initialized()` return value ends up being correct.